visualization/dashboard/src/MLIDashboard/StatisticsUtils.ts

```diff
@@ -1,24 +1,81 @@
+            const outcomeSubset = selectionArray.map(i => outcomes[i]);
+            return generateBinaryStats(outcomeSubset);
+        });
+    } if (modelType === ModelTypes.regression) {
+        const trueYs = jointDataset.unwrap(JointDataset.TrueYLabel);
+        const predYs = jointDataset.unwrap(JointDataset.PredictedYLabel);
+        const errors = jointDataset.unwrap(JointDataset.RegressionError);
+        return selectionIndexes.map(selectionArray => {
+            const trueYSubset = selectionArray.map(i => trueYs[i]);
+            const predYSubset = selectionArray.map(i => predYs[i]);
+            const errorsSubset = selectionArray.map(i => errors[i]);
+            return generateRegressionStats(trueYSubset, predYSubset, errorsSubset);
+        });
+    }
 }
 
-export const generateBinaryStats: (outcomes: number[]) => IBinaryStats = (outcomes: number[]): IBinaryStats => {
+const generateBinaryStats: (outcomes: number[]) => ILabeledStatistic[] = (outcomes: number[]): ILabeledStatistic[] => {
     const falseNegCount = outcomes.filter(x => x === ClassificationEnum.FalseNegative).length;
     const falsePosCount = outcomes.filter(x => x === ClassificationEnum.FalsePositive).length;
     const trueNegCount = outcomes.filter(x => x === ClassificationEnum.TrueNegative).length;
     const truePosCount = outcomes.filter(x => x === ClassificationEnum.TruePositive).length;
     const total = outcomes.length;
-    return {
-        accuracy: (truePosCount + trueNegCount) / total,
-        precision: truePosCount / (truePosCount + trueNegCount),
-        recall: truePosCount / (truePosCount + falseNegCount),
-        falsePositiveRate: falsePosCount / (trueNegCount + falsePosCount),
-        falseNegativeRate: falseNegCount / (truePosCount + falseNegCount)
-    }
-}
\ No newline at end of file
+    return [
+        {
+            label: localization.Statistics.accuracy,
+            stat: (truePosCount + trueNegCount) / total
+        },
+        {
+            label: localization.Statistics.precision,
+            stat: truePosCount / (truePosCount + trueNegCount)
+        },
+        {
+            label: localization.Statistics.recall,
+            stat: truePosCount / (truePosCount + falseNegCount)
+        },
+        {
+            label: localization.Statistics.fpr,
+            stat: falsePosCount / (trueNegCount + falsePosCount)
+        },
+        {
+            label: localization.Statistics.fnr,
+            stat: falseNegCount / (truePosCount + falseNegCount)
+        }
+    ];
+};
+
+const generateRegressionStats: (trueYs: number[], predYs: number[], errors: number[]) => ILabeledStatistic[] =
+    (trueYs: number[], predYs: number[], errors: number[]): ILabeledStatistic[] => {
+    const count = trueYs.length;
+    const residualSumOfSquares = errors.reduce((prev, curr) => {return prev + curr*curr}, 0);
+    const avgY = trueYs.reduce((prev, curr) => {return prev + curr}, 0)/count;
```

Comment:
very nitpicky: here and below, consistent spacing around operators

Reply:
+1 we really need to enable linting somehow

Reply:
you're right, I just like one space around multiply and divide, but it is asymetric, and now fixed.